aggregator/src/aggregator.rs

```diff
@@ -675,61 +688,6 @@ impl<C: Clock> Aggregator<C> {
             .await
     }
 
-    async fn task_aggregator_for(
-        &self,
-        task_id: &TaskId,
-    ) -> Result<Option<Arc<TaskAggregator<C>>>, Error> {
-        // TODO(#238): don't cache forever (decide on & implement some cache eviction policy). This
-        // is important both to avoid ever-growing resource usage, and to allow aggregators to
-        // notice when a task changes (e.g. due to key rotation).
-
-        // Step one: grab the existing entry for this task, if one exists. If there is no existing
-        // entry, write a new (empty) entry.
-        let cache_entry = {
-            // Unwrap safety: mutex poisoning.
-            let mut task_aggs = self.task_aggregators.lock().unwrap();
-            Arc::clone(
-                task_aggs
-                    .entry(*task_id)
-                    .or_insert_with(|| Arc::new(Mutex::default())),
-            )
-        };
-
-        // Step two: if the entry is empty, fill it via a database query. Concurrent callers
-        // requesting the same task will contend over this lock while awaiting the result of the DB
-        // query, ensuring that in the common case only a single query will be made for each task.
-        let task_aggregator = {
-            let mut cache_entry = cache_entry.lock().await;
-            if cache_entry.is_none() {
```

Comment:
@branlwyd I think we both overlooked this in https://github.com/divviup/janus/pull/3028. If `cache_entry.is_none()` is true, that means that we've cached "no task present" and we're going to go down the path of reloading it. That means that our check on 725 is redundant.

To work around this and allow the non-taskprov optimization, I track an additional variable `is_initialized` which lets the whole construct work more or less like a OnceCell.

Reply:
> I think we do need the check on L725; the point of that check is to remove None values so that we can't be DoS'ed by e.g. a malicious uploader trying to upload with random task IDs, causing us to store an unbounded amount of information.

Ah, makes sense, I missed that its purpose.